gdpr/anonymizers/fields.py

```diff
@@ -0,0 +1,125 @@
+import warnings
+from datetime import timedelta
+from typing import Any, Callable
+
+from django.core.exceptions import ImproperlyConfigured
+
+from gdpr.anonymizers.base import FieldAnonymizer
+
+
+class FunctionAnonymizer(FieldAnonymizer):
+    """
+    Use this field anonymization for defining in situ anonymization method.
+
+    Example:
+    ```
+    secret_code = FunctionFieldAnonymizer(lambda x: x**2)
+    ```
+    """
+    func = None
+
+    def __init__(self, func: Callable, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        if callable(func):
+            self.function = func
+        else:
+            raise ImproperlyConfigured("Supplied function is not callable.")
+
+    def get_anonymized_value(self, value):
+        return self.func(value)
+
+
+class PlaceHolderAnonymizer(FieldAnonymizer):
```

Comment:
jaký to má účel? Nestačí to tam nedávat?

Reply:
Smazáno